src/pages/home/report/ContextMenu/BaseReportActionContextMenu.tsx

```diff
@@ -0,0 +1,238 @@
+    });
+
+    /**
+     * Checks if user is anonymous. If true and the action doesn't accept for anonymous user, hides the context menu and
+     * shows the sign in modal. Else, executes the callback.
+     */
+    const interceptAnonymousUser = (callback: () => void, isAnonymousAction = false) => {
+        if (Session.isAnonymousUser() && !isAnonymousAction) {
+            hideContextMenu(false);
+
+            InteractionManager.runAfterInteractions(() => {
+                Session.signOutAndRedirectToSignIn();
+            });
+        } else {
+            callback();
+        }
+    };
+
+    useKeyboardShortcut(
+        CONST.KEYBOARD_SHORTCUTS.ENTER,
+        (event) => {
+            if (!menuItemRefs.current[focusedIndex]) {
+                return;
+            }
+
+            // Ensures the event does not cause side-effects beyond the context menu, e.g. when an outside element is focused
+            if (event) {
+                event.stopPropagation();
+            }
+
+            menuItemRefs.current[focusedIndex].triggerPressAndUpdateSuccess();
+            setFocusedIndex(-1);
+        },
+        {isActive: shouldEnableArrowNavigation},
+    );
+
+    return (
+        (isVisible || shouldKeepOpen) && (
+            <View
+                ref={contentRef}
+                style={wrapperStyle}
+            >
+                {filteredContextMenuActions.map((contextAction, index) => {
+                    const closePopup = !isMini;
+                    const payload: ContextMenuActionPayload = {
+                        reportAction: reportAction as ReportAction,
+                        reportID,
+                        draftMessage,
+                        selection,
+                        close: () => setShouldKeepOpen(false),
+                        openContextMenu: () => setShouldKeepOpen(true),
+                        interceptAnonymousUser,
+                    };
+
+                    if (contextAction.renderContent) {
+                        // make sure that renderContent isn't mixed with unsupported props
+                        if (__DEV__ && contextAction.icon != null) {
+                            throw new Error('Dev error: renderContent() and text/icon cannot be used together.');
+                        }
+
+                        return contextAction.renderContent(closePopup, payload);
+                    }
+
+                    return (
+                        <ContextMenuItem
+                            ref={(ref) => {
+                                menuItemRefs.current[index] = ref;
+                            }}
+                            // @ts-expect-error TODO: Remove this once ContextMenuItem (https://github.com/Expensify/App/issues/25056) is migrated to TypeScript.
+                            icon={contextAction.icon}
+                            text={contextAction.textTranslateKey ? translate(contextAction.textTranslateKey, {action: reportAction} as never) : undefined}
```

Comment:
Why do we need `as never` assertion here?

Reply:
Because of how translate function works, `textTranslateKey` is of type `TranslationPaths`, so TS doesn't know if under the hood string or function is used, and the second argument is never.

Reply:
As I can see from the translation files only the next keys support action param: `'reportActionContextMenu.editAction'`, `'reportActionContextMenu.deleteAction'` , `'reportActionContextMenu.deleteConfirmation'`
So I guess it should be possible to check if key is one of that keys
```
const {textTranslateKey} = contextAction;
const isKeyInActionUpdateKeys = textTranslateKey === 'reportActionContextMenu.editAction' || textTranslateKey === 'reportActionContextMenu.deleteAction'  || textTranslateKey === 'reportActionContextMenu.deleteConfirmation';
```
And then have it like
```
isKeyInActionUpdateKeys ? translate(textTranslateKey, {action: reportAction}) : translate(textTranslateKey)
```

Reply:
Nice idea, thanks!